packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -201,3 +648,843 @@ export const InputBoxButton = (props: InputBoxButtonProps): JSX.Element => {
+            tag,
+            closeTagIdx,
+            closeTagLength,
+            plainTextEndIndex,
+            startIndex,
+            oldPlainTextEndIndex,
+            mentionTagLength
+          );
+          result += resultValue;
+          lastProcessedHTMLIndex = htmlIndex;
+          processedChange = updatedChange;
+          // no need to handle plainTextSelectionEndIndex as the change will be added later
+          //proceed with the next calculations
+        } else if (tag.subTags !== undefined && tag.subTags.length !== 0 && tag.content !== undefined) {
+          // with subtags
+
+          // before the tag content
+          const stringBefore = htmlText.substring(lastProcessedHTMLIndex, tag.openTagIdx + tag.openTagBody.length);
+          lastProcessedHTMLIndex = closeTagIdx;
+
+          const [content] = updateHTML(
+            tag.content,
+            oldPlainText,
+            newPlainText,
+            tag.subTags,
+            startIndex - mentionTagLength,
+            oldPlainTextEndIndex - mentionTagLength,
+            '', // the part of the tag should be just deleted without processedChange update and change will be added after this tag
+            mentionTrigger
+          );
+          result += stringBefore + content;
+          //proceed with the next calculations
+        } else {
+          // no subtags
+          result += htmlText.substring(
+            lastProcessedHTMLIndex,
+            tag.openTagIdx + tag.openTagBody.length + startChangeDiff
+          );
+          lastProcessedHTMLIndex = closeTagIdx;
+          //proceed with the next calculations
+        }
+      } else if (startIndex < tag.plainTextBeginIndex && oldPlainTextEndIndex > plainTextEndIndex) {
+        console.log('updateHTML 1.4 result', result);
+        // the change starts before  the tag and finishes after it
+
+        // tag should be removed, no matter if there are subtags
+        // no need to save anything between lastProcessedHTMLIndex and closeTagIdx + closeTagLength
+        lastProcessedHTMLIndex = closeTagIdx + closeTagLength;
+        //proceed with the next calculations
+      } else if (startIndex === tag.plainTextBeginIndex && oldPlainTextEndIndex > plainTextEndIndex) {
+        console.log('updateHTML 1.5 result', result);
+        // the change starts in the tag and finishes after it
+        // tag should be removed, no matter if there are subtags
+        result += htmlText.substring(lastProcessedHTMLIndex, tag.openTagIdx);
+        // processedChange shouldn't be updated as it will be added after the tag
+        lastProcessedHTMLIndex = closeTagIdx + closeTagLength;
+        //proceed with the next calculations
```

Comment:
Space after //

Reply:
It will be fixed in the next PR with removing all console logs

Reply:
Updated